### PR TITLE
Fix link checks

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

Since the offending link is in [`whatsnew/1.0.rst`](https://github.com/SciTools/iris/blob/5555d455df4bfdbaec607e8a8da439ef8663dc1e/docs/src/whatsnew/1.0.rst), and is proving difficult to debug (only a problem on GHA), I'd say it's not worth fixing.

Moreover: it is adjacent to a slightly different WORKING link for ECMWF, so I have decided to remove the broken one rather than just ignore it during link checks - we get very little benefit by keeping it, but risk the link going silently stale in future.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

**Comment by trexfeathers** [comment]
Interestingly I can't replicate this when I run the same `nox --session linkcheck` locally, so it's something specific to GHA.

**Comment by trexfeathers** [comment]
Since the offending link is in [`whatsnew/1.0.rst`](https://github.com/SciTools/iris/blob/5555d455df4bfdbaec607e8a8da439ef8663dc1e/docs/src/whatsnew/1.0.rst), and is proving difficult to debug (only a problem on GHA), I'd say it's not worth fixing.

Moreover: it is adjacent to a slightly different WORKING link for ECMWF, so I have decided to remove the broken one rather than just ignore it during link checks - we get very little benefit by keeping it, but risk the link going silently stale in future.

File: docs/src/whatsnew/latest.rst [hunk 1]
```diff
@@ -69,6 +69,8 @@ This document explains the changes made to Iris for this release
 
 #. `@rcomer`_ clarified instructions for updating gallery tests. (:pull:`5100`)
 
+#. `@trexfeathers`_ fixed some broken links. (:pull:`5109`)
+
```
Comment (ESadek-MO) [inline docs/src/whatsnew/latest.rst:73]: Could maybe be slightly more specific?

**Review by ESadek-MO — CHANGES_REQUESTED** [review]
Hey @trexfeathers, this looks good! Would perhaps make the Whatsnew entry more specific, then I think this would be good to go!

File: docs/src/whatsnew/latest.rst (hunk 1 above)
Comment (trexfeathers) [reply docs/src/whatsnew/latest.rst]: 17303d78ba1660eb4f115c5932d355abc191b7f9

**Review by ESadek-MO — APPROVED** [review]

**Pull request merged**